src/linkify-matrix.ts

```diff
@@ -257,6 +261,8 @@ registerPlugin(Type.UserId, ({ scanner, parser, utils }) => {
     });
 });
 
+registerCustomProtocol("matrix");
```

Comment:
You can leave this in here but I don't think it's actually going to work until Linkify v4.0 https://github.com/Hypercontext/linkifyjs/issues/372

Reply:
yea I noticed it only works for `matrix://` but we need it to work for `matrix:r/...` as well

Reply:
This is now not part of this PR anymore (since this pr's main focus was to port to v3 I dont want to add any matrix: uri related parts here)